### PR TITLE
[IMP] pos: focus search input on ProductScreen when typing

### PR DESCRIPTION
**Pull request by nesma-neha**
- Added keydown event listener to focus the search input field when typing.

Task ID: 4070511

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/app/navbar/navbar.js
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/navbar.js:1]: why not use the `autofocus` prop ? https://github.com/odoo/odoo/blob/84cff1b20328c7dbde9ff759e8656afe7a4d40f2/addons/point_of_sale/static/src/app/generic_components/inputs/input/input.js#L23

Comment (nesma-neha) [reply addons/point_of_sale/static/src/app/navbar/navbar.js:1]: I didn't know it existed, will try it.

File: addons/point_of_sale/static/src/app/navbar/navbar.js [hunk 1]
```diff
@@ -46,8 +46,30 @@ export class Navbar extends Component {
         this.isBarcodeScannerSupported = isBarcodeScannerSupported;
         onMounted(async () => {
             this.isSystemUser = await user.hasGroup("base.group_system");
+            document.addEventListener("keydown", this.handleKeydown.bind(this));
         });
+        onWillUnmount(() => {
+            document.removeEventListener("keydown", this.handleKeydown.bind(this));
+        });
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/navbar.js:53]: could you use `useExternalListener` for this ?

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/navbar/navbar.js [hunk 2]
```diff
@@ -47,7 +47,26 @@ export class Navbar extends Component {
         onMounted(async () => {
             this.isSystemUser = await user.hasGroup("base.group_system");
         });
+        useExternalListener(document, "keydown", this.handleKeydown.bind(this));
     }
+
+    handleKeydown(event) {
+        if (
+            this.inputRef.el &&
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/navbar/navbar.js:55]: here you have to check that the input is actually rendered. Typing a key on the floor screen in restaurant for ex results in an error

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**